borg/platform.py

```diff
@@ -14,3 +18,86 @@ def acl_get(path, item, st, numeric_owner=False):
 
     def acl_set(path, item, numeric_owner=False):
         pass
+
+# GenericDirEntry, scandir_generic (c) 2012 Ben Hoyt
+# from the python-scandir package (3-clause BSD license, just like us, so no troubles here)
+# note: simplified version
+
+
+class GenericDirEntry(object):
+    __slots__ = ('name', '_stat', '_lstat', '_scandir_path', '_path')
+
+    def __init__(self, scandir_path, name):
+        self._scandir_path = scandir_path
+        self.name = name
+        self._stat = None
+        self._lstat = None
+        self._path = None
+
+    @property
+    def path(self):
+        if self._path is None:
+            self._path = os.path.join(self._scandir_path, self.name)
+        return self._path
+
+    def stat(self, follow_symlinks=True):
+        assert not follow_symlinks
+        return os.lstat(self.path)
+
+    def is_dir(self, follow_symlinks=True):
+        assert not follow_symlinks
+        try:
+            st = self.stat(False)
```

Comment:
same here, also self.stat made me think this would translate into stat, not lstat.

But even more importantly, why have these two methods at all? they would be racy and therefore unsafe.